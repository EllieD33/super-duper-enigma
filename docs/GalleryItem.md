# GalleryItem Component

## Description

The `GalleryItem` component renders an image inside a container, with optional overlay text. It supports different sizes and displays a loading skeleton or fallback image in case of errors or while loading.

## Props

### `GalleryItemProps`

| Prop             | Type                             | Required | Description                                                                                               |
| ---------------- | -------------------------------- | -------- | --------------------------------------------------------------------------------------------------------- |
| `imageUrl`       | `string`                         | ✅       | The URL of the image to display.                                                                          |
| `altText`        | `string`                         | ✅       | The alt text for the image.                                                                               |
| `imageId`        | `number`                         | ✅       | Unique identifier for the image.                                                                          |
| `size`           | `"small" \| "medium" \| "large"` | ❌       | Optional size of the image. Defaults to `"medium"`.                                                       |
| `overlayText`    | `string`                         | ❌       | Optional text to display over the image as an overlay.                                                    |
| `overlayDisplay` | `"hover" \| "static"`            | ❌       | Specifies when the overlay text should appear: `"hover"` for hover effect, `"static"` for always visible. |

## Behaviour

- The `GalleryItem` component renders an image inside a container.
- If the image is still loading, a skeleton loader will be displayed.
- If the image fails to load, a fallback image will be shown instead.
- Optional overlay text can be displayed, either always visible (`"static"`) or shown only on hover (`"hover"`).
- Supports multiple sizes of images: small, medium, and large.
- The component uses a lazy loading strategy for images to improve performance.

[Explore GalleryItem in Storybook](http://localhost:6006/?path=/docs/library-galleries-galleryitem--docs&args=error:&globals=viewport:largeMobile)

## Notes

- **Fallback Image**: If the `imageUrl` fails to load or has an error, a placeholder image (`https://placehold.co/600`) is shown instead.
- **Overlay Display**: When `overlayDisplay` is set to `"hover"`, the overlay text appears only when the user hovers over the image. If set to `"static"`, the text remains visible at all times.
- The **loading skeleton** is shown while the image is still loading, providing a better user experience during image load times.
- The `GalleryItem` component has a flexible design, adapting to different image sizes depending on the `size` prop.
