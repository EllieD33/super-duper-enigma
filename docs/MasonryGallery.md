# MasonryGallery Component

## Description

The `MasonryGallery` component displays images in a grid with a repeating pattern of large and small square images. The layout is designed to form a neat, structured grid, alternating between large and small squares, ensuring a visually balanced and consistent presentation of images.

## Props

### `MasonryGalleryProps`

| Prop     | Type                 | Required | Description                                                                                                     |
| -------- | -------------------- | -------- | --------------------------------------------------------------------------------------------------------------- |
| `images` | `GalleryItemProps[]` | âœ…       | An array of images to be displayed in the gallery. Each image must conform to the `GalleryItemProps` interface. |

### `GalleryItemProps`

See [GalleryItem Documentation](/docs/GalleryItem.md)

## Behaviour

- The `MasonryGallery` component arranges images in a neat grid, with alternating large and small square images.
- Each image is rendered using the `GalleryItem` component, ensuring consistent display of images.
- The grid pattern is formed by repeating large and small square images, creating a dynamic and visually appealing layout.
- Optionally, overlay text can be displayed on top of the image. The display mode of the overlay can be controlled with the `overlayDisplay` prop (`"hover"` or `"static"`).
- The gallery adapts dynamically to the available space, with the images arranged into the grid pattern.

[Explore MasonryGallery in Storybook](http://localhost:6006/?path=/story/library-galleries-masonrygallery--default&globals=viewport:largeTablet)

## Notes

- The `images` prop expects an array of objects adhering to the `GalleryItemProps` interface.
- The layout alternates between large and small square images, forming a structured and neat grid.
