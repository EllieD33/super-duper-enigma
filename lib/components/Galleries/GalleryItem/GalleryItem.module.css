.image {
    max-width: 100%;
    height: auto;
    border-radius: 15px;
    object-fit: cover;
}