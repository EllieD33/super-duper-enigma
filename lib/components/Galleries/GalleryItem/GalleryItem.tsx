import React, { ReactElement, useState } from "react";
import styles from "./GalleryItem.module.css";
import clsx from "clsx";

export interface GalleryItemProps {
    imageUrl: string;
    altText: string;
    imageId: number;
    size?: "small" | "medium" | "large";
    overlayText?: string;
}

const GalleryItem = ({
    imageUrl,
    altText,
    imageId,
    size = "medium",
    overlayText,
}: GalleryItemProps): ReactElement => {
    const [isLoading, setIsLoading] = useState<boolean>(true);
    const [hasError, setHasError] = useState<boolean>(false);

    const galleryItemStyles = clsx(styles.image, styles[size]);
    const skeletonItemStyles = clsx(styles.SkeletonItem, styles[size]);
    const overlayContainerStyles = overlayText
        ? clsx(styles.overlayContainer)
        : "";
    const overlayContentStyles = overlayText ? clsx(styles.overlayContent) : "";

    const fallbackImage = "https://placehold.co/600";

    const handleImageLoad = () => {
        setIsLoading(false);
    };

    const handleImageError = () => {
        setIsLoading(false);
        setHasError(true);
    };

    return (
        <div id={`gallery-item-${imageId}`} className={styles.container}>
            <div className={overlayContainerStyles}>
                <img
                    className={galleryItemStyles}
                    src={hasError ? fallbackImage : imageUrl}
                    alt={hasError ? "image not found" : altText}
                    loading="lazy"
                    onLoad={handleImageLoad}
                    onError={handleImageError}
                    data-testid="galleryItemImage"
                />
                {overlayText && (
                    <div className={overlayContentStyles}>{overlayText}</div>
                )}
            </div>
            {isLoading && !hasError && (
                <div
                    className={skeletonItemStyles}
                    data-testid="skeleton"
                ></div>
            )}
        </div>
    );
};

export default GalleryItem;
