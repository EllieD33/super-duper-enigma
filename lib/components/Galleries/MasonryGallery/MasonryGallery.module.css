.container {
    display: grid;
    gap: 8px;
    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
    grid-auto-flow: dense;
}

@media (min-width: 480px) {
    .container {
        grid-template-columns: repeat(2, 1fr);
    }
}

@media (min-width: 768px) {
    .container {
        grid-template-columns: repeat(4, 1fr);
    }
    .container > *:nth-child(20n + 1),
    .container > *:nth-child(20n + 4),
    .container > *:nth-child(20n + 13),
    .container > *:nth-child(20n + 14) {
        grid-column: span 2;
        grid-row: span 2;
    }
}

@media (min-width: 1200px) {
    .container {
        grid-template-columns: repeat(6, 1fr);
    }
    .container > *:nth-child(60n + 5),
    .container > *:nth-child(60n + 7),
    .container > *:nth-child(60n + 21),
    .container > *:nth-child(60n + 34),
    .container > *:nth-child(60n + 40),
    .container > *:nth-child(60n + 54) {
        grid-column: span 2;
        grid-row: span 2;
    }
}
