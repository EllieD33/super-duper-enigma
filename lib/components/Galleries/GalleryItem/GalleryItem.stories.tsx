import type { Meta, StoryObj } from "@storybook/react";
import GalleryItem, { GalleryItemProps } from "./GalleryItem";

const meta = {
    title: "Library/Galleries/GalleryItem",
    component: GalleryItem,
    parameters: {
        layout: "centered",
    },
    tags: ["autodocs"],
    argTypes: {},
} satisfies Meta<typeof GalleryItem>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        imageId: 1,
        imageUrl: "https://cdn.pixabay.com/photo/2023/08/08/15/01/flower-8177578_640.jpg",
        altText: "white flowers wth dark green leaves behind them",
    } as GalleryItemProps,
};
