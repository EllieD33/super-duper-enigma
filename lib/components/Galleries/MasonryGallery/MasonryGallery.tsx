import React, { ReactElement } from "react";
import GalleryItem, { GalleryItemProps } from "../GalleryItem/GalleryItem";
import styles from "./MasonryGallery.module.css";

export interface MasonryGalleryProps {
    images: GalleryItemProps[];
}

const MasonryGallery = ({ images }: MasonryGalleryProps): ReactElement => {
    if (!images || images.length === 0) {
        return <></>;
    }

    return (
        <div
            className={styles.container}
            data-testid="masonry-gallery-container"
        >
            {images &&
                images.map((image) => (
                    <div key={image.imageId}>
                        <GalleryItem {...image} />
                    </div>
                ))}
        </div>
    );
};

export default MasonryGallery;
