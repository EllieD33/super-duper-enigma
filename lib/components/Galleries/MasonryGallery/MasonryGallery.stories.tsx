import type { Meta, StoryObj } from "@storybook/react";
import MasonryGallery, { MasonryGalleryProps } from "./MasonryGallery";
import mockImages from "./testData/testData";

const meta = {
    title: "Library/Galleries/MasonryGallery",
    component: MasonryGallery,
    parameters: {
        layout: "centered",
    },
    tags: ["autodocs"],
    argTypes: {},
} satisfies Meta<typeof MasonryGallery>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        images: mockImages,
    } as MasonryGalleryProps,
};
