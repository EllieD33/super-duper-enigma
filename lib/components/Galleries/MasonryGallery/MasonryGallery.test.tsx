import React from "react";
import { render, screen } from "@testing-library/react";
import MasonryGallery, { MasonryGalleryProps } from "./MasonryGallery";
import { GalleryItemProps } from "../GalleryItem/GalleryItem";
import mockImages from "./testData/testData";

const defaultProps: MasonryGalleryProps = {
    images: mockImages,
};

describe("MasonryGallery", () => {
    it("should render the default component", () => {
        const { container } = render(<MasonryGallery {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });

    it("should render the correct number of gallery items", () => {
        render(<MasonryGallery {...defaultProps} />);
        const images = screen.getAllByTestId("galleryItemImage");
        expect(images).toHaveLength(mockImages.length);
    });

    it("should pass the correct props to GalleryItem", () => {
        render(<MasonryGallery {...defaultProps} />);
        mockImages.forEach((image: GalleryItemProps) => {
            const imgElement = screen.getByAltText(image.altText);
            expect(imgElement).toHaveAttribute("src", image.imageUrl);
        });
    });

    it("should apply the correct container class", () => {
        const { container } = render(<MasonryGallery {...defaultProps} />);
        expect(container.firstChild).toHaveClass("container");
    });

    it("should render nothing when images array is empty", () => {
        render(<MasonryGallery images={[]} />);
        const container = screen.queryByTestId("masonry-gallery-container");
        expect(container).not.toBeInTheDocument();
    });
});
