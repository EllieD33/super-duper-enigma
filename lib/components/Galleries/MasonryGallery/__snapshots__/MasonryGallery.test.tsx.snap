// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasonryGallery should render the default component 1`] = `
<div>
  <div
    class="container"
    data-testid="masonry-gallery-container"
  >
    <div>
      <div
        class="container"
        data-testid="gallery-item-1"
      >
        <div>
          <img
            alt="image1 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/08/08/15/01/flower-8177578_640.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-2"
      >
        <div>
          <img
            alt="image2 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/04/21/11/51/flower-7941764_640.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-3"
      >
        <div>
          <img
            alt="image3 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/01/17/17/22/bee-8515123_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-4"
      >
        <div>
          <img
            alt="image4 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/04/11/22/08/flower-7918323_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-5"
      >
        <div>
          <img
            alt="image5 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/06/04/12/00/rose-8039502_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-6"
      >
        <div>
          <img
            alt="image6 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/07/28/20/36/flower-8155951_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-7"
      >
        <div>
          <img
            alt="image7 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/06/27/16/30/flower-8092680_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-8"
      >
        <div>
          <img
            alt="image8 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/04/21/12/18/flower-7941814_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-9"
      >
        <div>
          <img
            alt="image9 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/03/21/01/31/flower-7866187_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-10"
      >
        <div>
          <img
            alt="image10 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/10/18/08/17/rose-8323259_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-11"
      >
        <div>
          <img
            alt="image11 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/07/23/08/46/flower-8144644_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-12"
      >
        <div>
          <img
            alt="image12 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/04/10/15/12/succulent-7914023_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-13"
      >
        <div>
          <img
            alt="image13 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/01/09/12/49/ferns-7707348_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-14"
      >
        <div>
          <img
            alt="image14 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/11/02/15/58/flower-8360946_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-15"
      >
        <div>
          <img
            alt="image15 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/03/15/18/53/magnolia-flower-8635583_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-16"
      >
        <div>
          <img
            alt="image16 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/11/13/01/30/flower-8384359_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-17"
      >
        <div>
          <img
            alt="image17 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/03/04/16/07/winter-8612635_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-18"
      >
        <div>
          <img
            alt="image18 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/01/04/19/40/grass-7697505_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-19"
      >
        <div>
          <img
            alt="image19 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/10/02/05/29/flowers-8288492_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-20"
      >
        <div>
          <img
            alt="image20 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/08/31/21/52/houseleek-8225994_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-21"
      >
        <div>
          <img
            alt="image21 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/04/02/09/15/sempervivum-7894111_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-22"
      >
        <div>
          <img
            alt="image22 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/03/19/16/15/butterfly-7862893_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-23"
      >
        <div>
          <img
            alt="image23 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/09/14/22/16/fern-8253836_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-24"
      >
        <div>
          <img
            alt="image24 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/07/23/06/57/leaf-8144517_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-25"
      >
        <div>
          <img
            alt="image25 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/02/27/00/13/heliconia-8599119_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-26"
      >
        <div>
          <img
            alt="image26 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/05/15/07/59/flowers-8763039_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-27"
      >
        <div>
          <img
            alt="image27 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/07/08/19/28/leaves-8115077_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-28"
      >
        <div>
          <img
            alt="image28 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/09/22/07/02/red-8268266_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-29"
      >
        <div>
          <img
            alt="image29 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/02/16/14/20/snail-8577681_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-30"
      >
        <div>
          <img
            alt="image30 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2024/01/18/16/54/leaves-8517219_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-31"
      >
        <div>
          <img
            alt="image31 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/04/27/14/27/butterfly-7954767_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
    <div>
      <div
        class="container"
        data-testid="gallery-item-32"
      >
        <div>
          <img
            alt="image32 alt text"
            class="image"
            data-testid="galleryItemImage"
            loading="lazy"
            src="https://cdn.pixabay.com/photo/2023/11/13/00/47/cactus-8384331_960_720.jpg"
          />
        </div>
        <div
          class="SkeletonItem"
          data-testid="skeleton"
        />
      </div>
    </div>
  </div>
</div>
`;
